### PR TITLE
remove ErrorHandlingMethod in favour of NoDataStrategy

### LINKED ISSUE #1036: Only support one way of raising exceptions for no data.
### Kind of request

Changing existing functionality

### Enhancement Description

Currently there are 2 implementations for checking for no data: `NoDataStrategy` and `handle_error_method`. We should refactor to use just one and keep it that way.

### Use case

Maintainability

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by savente93**
## Issue addressed
Fixes #1036 
## Explanation
HandleNodataStrategy is much more widely used in the driver side, while ErrorHandlingMethod was only really really used for stac. Thefore we're deprecating the latter in favour of the former

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `main`
- [x] Tests & pre-commit hooks pass
- [ ] Updated documentation
- [ ] Updated changelog.rst

## Data/Catalog checklist
- [ ] `data/catalogs/predefined_catalogs.yml` has not been modified.
- [ ] None of the old `data_catalog.yml` files have been chagned
- [ ] `data/chagnelog.rst` has been updated
- [ ] new file uses `LF` line endings (done automatically if you used `update_versions.py`)
- [ ] New file has been tested locally
- [ ] Tests have been added using the new file in the test suite

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: tests/data_catalog/test_data_catalog.py [hunk 1]
```diff
@@ -713,16 +694,6 @@ def test_to_stac(self, data_catalog: DataCatalog):
...
 
-        outcome = cast(
-            StacCatalog, source.to_stac_catalog(on_error=ErrorHandleMethod.RAISE)
-        )
-
-        assert raster_stac_catalog.to_dict() == outcome.to_dict()  # type: ignore
-        source.metadata.crs = (
-            -3.14
-        )  # manually create an invalid adapter by deleting the crs
-        assert source.to_stac_catalog(on_error=ErrorHandleMethod.SKIP) is None
```
Comment (Jaapel) [inline tests/data_catalog/test_data_catalog.py:724]: Why remove the assertions?

File: tests/data_catalog/sources/test_geo_dataframe_source.py [hunk 2]
```diff
@@ -122,35 +115,3 @@ def test_geodataframe_unit_attrs(self, artifact_data: DataCatalog):
...
-
-        gdf_stac_catalog.add_item(gds_stac_item)
-        outcome = cast(
-            StacCatalog, adapter.to_stac_catalog(on_error=ErrorHandleMethod.RAISE)
-        )
-        assert gdf_stac_catalog.to_dict() == outcome.to_dict()  # type: ignore
-        adapter.metadata.crs = (
-            -3.14
-        )  # manually create an invalid adapter by deleting the crs
-        assert adapter.to_stac_catalog(on_error=ErrorHandleMethod.SKIP) is None
```
Comment (Jaapel) [inline tests/data_catalog/sources/test_geo_dataframe_source.py:156]: Same here

**Review by Jaapel — CHANGES_REQUESTED** [review]
You removed a few tests, which do not only check the previous `on_error` behaviour, but also test the methods. Why is that?

File: tests/data_catalog/test_data_catalog.py (hunk 1 above)
Comment (savente93) [reply tests/data_catalog/test_data_catalog.py:724]: I didn't really read them properly and thought they were mostly making asserions about the coerce option which no longer exists, I'll put them back

**Review by Jaapel — APPROVED** [review]

**Pull request merged**